qiskit/extensions/unitary.py

```diff
@@ -89,6 +92,10 @@ def transpose(self):
 
     def _define(self):
         """Calculate a subcircuit that implements this unitary."""
+        if self.num_qubits == 1:
+            q = QuantumRegister(1, "q")
+            angles = euler_angles_1q(Operator(self).data)
```

Comment:
You should change this to
```python
angles = euler_angles_1q(self.to_matrix())
```
to avoid having to explicitly import Operator class which could lead to cyclic imports (the Operator class imports UnitaryGate for its `to_instruction` method).

Reply:
done